### ISSUE TITLE
Imviz: AsdfInFits removal from ASDF 3.0

### ISSUE THREAD

**Issue by braingram** [spacetelescope/jdaviz#1980]
- [ ] Remove the warning ignore here (added in #2021):

https://github.com/spacetelescope/jdaviz/blob/71a514c1cf7741874d2770f34c5d0ff9045038a7/setup.cfg#L105

- [ ] Can we also revert https://github.com/spacetelescope/jdaviz/pull/2035 ?

The ability to read AsdfInFits files is being moved from [ASDF](https://github.com/asdf-format/asdf) into [stdatamodels](https://github.com/spacetelescope/stdatamodels) (in part to consolidate non-pipeline code for jwst). The current plan for this move is:

- include AsdfInFits in stdatamodels: https://github.com/spacetelescope/stdatamodels/pull/110
- ~~release a stable version of stdatamodels~~ EDIT: see comment below on stdatamodels >1.0
- drop AsdfInFits support in ASDF 3.0: https://github.com/asdf-format/asdf/pull/1288

We do not have a estimate for when a stable version of stdatamodels or ASDF 3.0 will be released. We are planning to release ASDF 2.15 (which will support AsdfInFits) in the next few weeks which we will continue to support for at least 1 year (please see our proposed [release and support process](https://github.com/asdf-format/asdf/pull/1323)).

To summarize, at some point in the next year or so the following uses of AsdfInFits in jdaviz will need to be updated to use stdatamodels.
https://github.com/spacetelescope/jdaviz/blob/20b224e616f61f0fd4245807560d7025aefc6867/jdaviz/configs/imviz/plugins/parsers.py#L232
https://github.com/spacetelescope/jdaviz/blob/f257d716a101b59807b4dc05a1e4dd19c020d30b/jdaviz/configs/imviz/tests/test_parser.py#L230

I mentioned this to @pllim and am happy to discuss alternatives, suggestions, comments on this process and hope to make it as painless as possible.

[🐱](https://jira.stsci.edu/browse/JDAT-3056)

Also see https://github.com/astropy/specutils/issues/1006

**Comment by braingram** [comment]
To provide a minor update to this issue, a version of stdatamodels with AsdfInFits support has been released:
https://pypi.org/project/stdatamodels/1.1.0/
with documentation on reading and writing AsdfInFits files: https://stdatamodels.readthedocs.io/en/latest/asdf_in_fits.html
